### DISCUSSION TITLE
I don't know why cv2.rectangle causes errors

### DISCUSSION THREAD

**Discussion by leejoohi** [community/community#58842] [Programming Help]
### Select Topic Area

Question

### Body

`from _testinternalcapi import get_config
from django.shortcuts import render
from django.http import StreamingHttpResponse
import yolov8

import cv2
from collections import Counter

from PIL import Image as im
from ultralytics import YOLO
from ultralytics.yolo.utils.plotting import Annotator
from ultralytics.yolo.utils.torch_utils import select_device

# Create your views here.
def index(request):
    return render(request, 'index.html')

model = YOLO('best.pt')

def stream():
    cap = cv2.VideoCapture(0)
    while True:
        ret, frame = cap.read()
        if not ret:
            print("Error: failed to capture image")
            break

        results = model(frame, augment=True)
        cv2.imwrite('demo.jpg', frame)

        colors = ['brown', 'green', 'orange', 'purple', 'red', 'yellow']
        cs = ['brown', 'green', 'orange', 'purple', 'red', 'yellow']
        cal = []
        try:
            tensor_list = results[0].boxes.data
            detection = tensor_list.tolist()
            total_product_count = len(detection)
            if len(detection) == 0:
                print("Detected 0 Products")
                return "Unknow", "Found 0"
            else:
                for det in detection:
                    x, y, w, h = int(det[0]), int(det[1]), int(det[2]), int(det[3])
                    confidence = det[4]
                    cls = det[5]
                    cal.append(str(colors[int(cls)]))
                    cv2.rectangle('demo.jpg', (x, y), (w, h), (0, 255, 0), 2)
                    cv2.putText('demo.jpg', colors[int(cls)], (x, y - 10), fontFace=cv2.FONT_HERSHEY_SIMPLEX, fontScale=0.6,
                                color=(255, 255, 255), thickness=2)

                count_products = counter(cal)
                products = dict_to_str(count_products)
                cv2.imwrite('demo.jpg', frame)
                return products, total_product_count
        except Exception as e:
            print(e)
            return "ERORR", "Unknown"

        yield (b'--frame\r\n'
               b'Content-Type: image/jpeg\r\n\r\n' + open('demo.jpg', 'rb').read() + b'\r\n')

def video_feed(request):
    return StreamingHttpResponse(stream(), content_type='multipart/x-mixed-replace; boundary=frame')`

**
We created a guest house detection learning model that detects colors through yolov8m. In this state, I wrote a code after watching YouTube for real-time guest house detection, but it doesn't work from cv2. correctangle. The error code is as follows.

**
OpenCV(4.7.0) :-1: error: (-5:Bad argument) in function 'rectangle'
> Overload resolution failed:
>  - img is not a numpy array, neither a scalar
>  - Expected Ptr<cv::UMat> for argument 'img'
>  - img is not a numpy array, neither a scalar
>  - Expected Ptr<cv::UMat> for argument 'img'

**
Please let me know which one I should modify.

**Comment by Hamza-Bin-Aamir** [comment]
### The issue
`cv2.rectangle()` does not take the name of the image file, it needs (as the error says) a numpy array
### How to fix it
you can achieve this by moving the `cv2.imwrite()` function **below** the try except blocks 
and changing the `cv2.rectangle()` and `cv2.putText()` functions to:

```
cv2.rectangle(frame , (x, y), (w, h), (0, 255, 0), 2) 

cv2.putText(frame , colors[int(cls)], (x, y - 10), fontFace=cv2.FONT_HERSHEY_SIMPLEX, 
    fontScale=0.6, color=(255, 255, 255), thickness=2)
```
